### ISSUE TITLE
Typing problems in groupby method

### ISSUE THREAD

**Issue by DraftProducts** [prisma/prisma#10915]
### Bug description

For `<Proposal>.groupBy` method, the `_count` value has a invalid type, it should only be a boolean as we can see in the doc :
> ![image](https://user-images.githubusercontent.com/29425008/147853070-9233f118-e142-49f7-9b63-0efa7326743a.png)
https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#groupby

![image](https://user-images.githubusercontent.com/29425008/147852854-4df81fda-4a87-4f34-aa3e-0928644fdd19.png)

This error let the developer make this type of code: 
![image](https://user-images.githubusercontent.com/29425008/147852897-1fffc500-3b13-4464-8b51-48aba4dc2865.png)

Witch let thinking that it would only return the count of elements which is in accord with the object definition (only merged proposals).

### How to reproduce

1. Use `<Model>.groupBy` method with `_count` property

### Expected behavior

Only have the optional typing as written in the doc (which make more sense)

### Prisma information

```js
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model Proposal {
  id            Int           @id @default(autoincrement())
  merged        Boolean       @default(false)
  
  // table content removed

  @@map("proposals")
}
```

### Environment & setup

- OS: Docker Node Image based on Ubuntu
- Database: PostgreSQL
- Node.js version: latest

### Prisma Version

```
prisma                  : 3.7.0
@prisma/client          : 3.7.0
Current platform        : debian-openssl-1.1.x
Query Engine (Node-API) : libquery-engine 8746e055198f517658c08a0c426c7eec87f5a85f (at node_modules/@prisma/engines/libquery_engine-debian-openssl-1.1.x.so.node)
Migration Engine        : migration-engine-cli 8746e055198f517658c08a0c426c7eec87f5a85f (at node_modules/@prisma/engines/migration-engine-debian-openssl-1.1.x)
Introspection Engine    : introspection-core 8746e055198f517658c08a0c426c7eec87f5a85f (at node_modules/@prisma/engines/introspection-engine-debian-openssl-1.1.x)
Format Binary           : prisma-fmt 8746e055198f517658c08a0c426c7eec87f5a85f (at node_modules/@prisma/engines/prisma-fmt-debian-openssl-1.1.x)
Default Engines Hash    : 8746e055198f517658c08a0c426c7eec87f5a85f
Studio                  : 0.445.0
```

**Comment by aqrln** [comment]
Hi @DraftProducts and thank you for taking your time to prepare the report and the linked PR!

As far as I can see, there's no bug and the type is actually correct, but the documentation could be improved here. You may, in fact, pass an object to the `_count` argument of `groupBy`, similar to how it works in the standalone `count` operation: https://www.prisma.io/docs/concepts/components/prisma-client/aggregation-grouping-summarizing#count-non-null-field-values

Now where the confusion comes from is since your field is a boolean one, `_count: { merged: true }` might read like it means "count all records where `merged` is `true`", but that's not what happens. What this actually means is that Prisma will count the records where `merged` is not `NULL`. You can also pass `_all` in that object to count all records in addition to counting instances of non-null fields.

In other words, if you make the `merged` field optional in your example and run this:

```ts
  await prisma.user.create({
    data: {
      proposals: {
        createMany: {
          data: [
            { merged: true },
            { merged: false },
            { merged: null }
          ]
        }
      }
    }
  })

  const proposals = await prisma.proposal.groupBy({
    by: ['userId'],
    having: {
      userId: {
        not: null
      }
    },
    _count: {
      merged: true,
      _all: true
    }
  })
```

You will get this as the result:

```js
[ { _count: { merged: 2, _all: 3 }, userId: 3 } ]
```

I'll go ahead and close this issue, feel free to reopen it if I missed anything.

**Comment by DraftProducts** [comment]
Sure, thx for the review and the time that you invest in Prisma, it's really fantastic!
